src/main/java/io/github/essencepowered/essence/config/KitsConfig.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This file is part of Essence, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.essencepowered.essence.config;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.reflect.TypeToken;
+import io.github.essencepowered.essence.internal.KitInventory;
+import ninja.leaping.configurate.commented.CommentedConfigurationNode;
+import ninja.leaping.configurate.commented.SimpleCommentedConfigurationNode;
+import ninja.leaping.configurate.hocon.HoconConfigurationLoader;
+import ninja.leaping.configurate.objectmapping.ObjectMappingException;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.item.inventory.Inventory;
+import org.spongepowered.api.item.inventory.ItemStack;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class KitsConfig extends AbstractConfig<CommentedConfigurationNode, HoconConfigurationLoader> {
+
+    private Map<String, KitInventory> kitNodes;
+
+    public KitsConfig(Path file) throws IOException, ObjectMappingException {
+        super(file);
+    }
+
+    @Override
+    public void load() throws IOException, ObjectMappingException {
+        super.load();
+
+        if (kitNodes == null) {
+            kitNodes = Maps.newHashMap();
+        }
+
+        kitNodes.clear();
+
+        node.getChildrenMap().forEach((k, v) -> {
+            try {
+                kitNodes.put(String.valueOf(k), v.getValue(TypeToken.of(KitInventory.class), new KitInventory()));
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        });
+    }
+
+    @Override
+    public void save() throws IOException, ObjectMappingException {
+        node = SimpleCommentedConfigurationNode.root();
+        kitNodes.forEach((k, v) -> node.getNode(k).setValue(v));
+        super.save();
+    }
+
+    @Override
+    protected HoconConfigurationLoader getLoader(Path file) {
+        return HoconConfigurationLoader.builder().setPath(file).build();
+    }
+
+    @Override
+    protected CommentedConfigurationNode getDefaults() {
+        CommentedConfigurationNode ccn = SimpleCommentedConfigurationNode.root();
+        ccn.getNode("kits").setComment("Contains all Kits.");
```

Comment:
This won't be needed, you're going to use the root node. Just get rid of `getNode("kits")` - the comment is useful (but please make it translatable using the properties file!)